File: python/GateThree/task_one_test.py
import task_one
from unittest import TestCase

class TaskOneTest(TestCase):
    def test_that_length_is_reduced(self):
        numbers = [10,11,12,13,14]
        number = 2
        result = task_one.reduce_length(numbers,number)
        self.assertEqual(result,[10,11])

    def test_that_empty_list_is_filled(self):
        number = 4
        result = task_one.fill_empty_list(number)
        self.assertEqual(result,[-1,-1,-1,-1])

    def test_that_short_list_is_filled_up_with_default_value(self):
        numbers = [10,11,12]
        number = 5
        result = task_one.add_default_value_to_list(numbers,number)
        self.assertEqual(result,[10,11,12,-1,-1])

File: java/GateThree/TaskOneTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;



public class TaskOneTest{
    @Test
    public void testTheLengthOfTheArray(){
    TaskOne testing = new TaskOne();
    int [] numbers = {10,11,12,13,14};
    int number = 2;
    int [] actualResult = testing.reduceLength(numbers,number);
    int [] expectedResult = {10,11};
    assertArrayEquals(actualResult, expectedResult);


    }

    @Test
    public void testTheDefaultNumOfTheArray(){
    TaskOne testing = new TaskOne();
    int [] numbers = {10,11,12};
    int number = 5;
    int [] actualResult = testing.fillDefault(numbers,number);
    int [] expectedResult = {10,11,12,-1,-1};
    assertArrayEquals(actualResult, expectedResult);

    }

    @Test
    public void testTheFillingNumOfTheArray(){
    TaskOne testing = new TaskOne();
    int number = 5;
    int [] numbers =  new int [number];
   
    int [] actualResult = testing.fillEmptyArray(number);
    int [] expectedResult = {-1,-1,-1,-1,-1};
    assertArrayEquals(actualResult, expectedResult);

    }









}
